### DISCUSSION TITLE
MacOS: is nested virtualization possible?

### DISCUSSION THREAD

**Discussion by volisoft** [utmapp/UTM#2606] [UTM Help (macOS)]
Is it possible to enable KVM kernel module in Ubuntu guest given the setup below:
```
Host: MacOS (BigSur 11.4, Apple Silicon)
Guest: Ubuntu ARM64
```

**Comment by lgrachov** [comment]
No. UTM does not support nested virtualisation (at least for now).

**Reply by daemontroll** [reply]
I see the beta has support for Linux vms, but what about Mac vms running Sequoia?

**Reply by lgrachov** [reply]
> I see the beta has support for Linux vms, but what about Mac vms running Sequoia?

I have tried installing UTM on a macOS VM running Sonoma, and only MacOS 9.2.1 worked and ReactOS (almost) worked.

See #6480.

**Comment by daemontroll** [comment]
So now that UTM offers nested virtualization on Linux guests, when can we expect that feature for MacOS guests?  Apple seems to support it in, https://developer.apple.com/documentation/virtualization/vzgenericplatformconfiguration/4360553-isnestedvirtualizationsupported.

**Reply by osy** [reply]
What you linked to is for Linux only.

**Reply by daemontroll** [reply]
I see that now, here is an article that seems to indicate that in order for that to work on a mac with apple silicon you need to enable virtualization extensions.  https://developer.apple.com/documentation/hypervisor/

**Reply by osy** [reply]
Yes, as someone who has worked with that framework for the past few years, I am very familiar with what it says.

**Reply by thomasjm** [reply]
So is there any timeline for nested virtualization for macOS guests?

**Reply by osy** [reply]
You need to ask apple
